cmd/lotus-shed/provider.go

```diff
@@ -0,0 +1,183 @@
+			Flags: []cli.Flag{
+				&cli.StringFlag{
+					Name:    FlagMinerRepo,
+					Aliases: []string{FlagMinerRepoDeprecation},
+					EnvVars: []string{"LOTUS_MINER_PATH", "LOTUS_STORAGE_PATH"},
+					Value:   "~/.lotusminer",
+					Usage:   fmt.Sprintf("Specify miner repo path. flag(%s) and env(LOTUS_STORAGE_PATH) are DEPRECATION, will REMOVE SOON", FlagMinerRepoDeprecation),
+				},
+				&cli.StringFlag{
+					Name:    "layer",
+					Aliases: []string{"l"},
+					Usage:   "The layer name for this data push. 'base' is recommended for single-miner setup. Overwrites.",
+				},
+			},
+			Action: fromMiner,
+		},
+	},
+}
+
+const (
+	FlagMinerRepo = "miner-repo"
+)
+
+const FlagMinerRepoDeprecation = "storagerepo"
+
+func fromMiner(cctx *cli.Context) error {
+	ctx := context.Background()
+
+	r, err := repo.NewFS(cctx.String(FlagMinerRepo))
+	if err != nil {
+		return err
+	}
+
+	ok, err := r.Exists()
+	if err != nil {
+		return err
+	}
+
+	if !ok {
+		return fmt.Errorf("repo not initialized")
+	}
+
+	lr, err := r.LockRO(repo.StorageMiner)
+	if err != nil {
+		return fmt.Errorf("locking repo: %w", err)
+	}
+	defer func() { _ = lr.Close() }()
+
+	cfgNode, err := lr.Config()
+	if err != nil {
+		return fmt.Errorf("getting node config: %w", err)
+	}
+	smCfg := cfgNode.(*config.StorageMiner)
+
+	db, err := harmonydb.NewFromConfig(smCfg.HarmonyDB)
+	if err != nil {
+		return fmt.Errorf("could not reach the database. Ensure the Miner config toml's HarmonyDB entry"+
+			" is setup to reach Yugabyte correctly: %w", err)
+	}
+
+	var titles []string
+	err = db.Select(ctx, &titles, `SELECT title FROM harmony_config`)
+	if err != nil {
+		return fmt.Errorf("miner cannot reach the db. Ensure the config toml's HarmonyDB entry"+
+			" is setup to reach Yugabyte correctly: %s", err.Error())
+	}
+	name := cctx.String("layer")
+	if name == "" {
+		name = fmt.Sprintf("mig%d", len(titles))
+	}
+	msg := "Layer " + name + ` created. `
+
+	// Copy over identical settings:
+
+	buf, err := os.ReadFile(path.Join(lr.Path(), "config.toml"))
+	if err != nil {
+		return fmt.Errorf("could not read config.toml: %w", err)
+	}
+	var lpCfg config.LotusProviderConfig
+	_, err = toml.Decode(string(buf), &lpCfg)
+	if err != nil {
+		return fmt.Errorf("could not decode toml: %w", err)
+	}
+
+	// Populate Miner Address
+	sm, cc, err := cliutil.GetStorageMinerAPI(cctx)
+	if err != nil {
+		return fmt.Errorf("could not get storageMiner API: %w", err)
+	}
+	defer cc()
+	addr, err := sm.ActorAddress(ctx)
+	if err != nil {
+		return fmt.Errorf("could not read actor address: %w", err)
+	}
+
+	lpCfg.Addresses.MinerAddresses = []string{addr.String()}
+
+	// TODO Apis.StorageSecret  storage repo and reading the rpc secret
+	//lr.??
+
+	// Populate API Key
+	_, header, err := cliutil.GetRawAPI(cctx, repo.FullNode, "v0")
+	if err != nil {
+		return fmt.Errorf("cannot read API: %w", err)
+	}
+
+	lpCfg.Apis.FULLNODE_API_INFO = []string{header.Get("Authorization")[7:]}
```

Comment:
TBH I'd rename this config from `FULLNODE_API_INFO` to something more consistent with the rest of the values. `ChainApiInfo` sounds decent to me.